webpack/assets/javascripts/react_app/common/I18n.js

```diff
@@ -61,7 +59,8 @@ const mergeLocaleData = locale => {
     Object.entries(translations).forEach(([source, translated]) => {
       if (
         result[source] === undefined ||
-        (result[source] === [''] && translated !== [''])
+        (JSON.stringify(result[source]) === JSON.stringify(['']) &&
+          JSON.stringify(translated) !== JSON.stringify(['']))
```

Comment:
Why is this necessary?

Reply:
Maybe because in JS `[''] === [''] // false` ?..

Reply:
/me sighs

Of course. This is evaluated each time the application is loaded, correct? Which may or may not mean every time the user navigates to a page?

Reply:
That makes me think of https://www.destroyallsoftware.com/talks/wat

Reply:
(it took me a few years to agree to move from C to JS after seeing this video for the first time)

Reply:
> it ran 4k times in opening the login page.

That's quite a lot. I'm pretty sure there's a more efficient way of determining whether an array contains a single element (which is an empty string) rather than dumping both sides into json and comparing those. If this got run only occasionally then I'd be willing to let this slide, but this is too frequent

Reply:
I can check the first element `result[source].?[0] === ''`?

Reply:
BTW its runs for every translated string

Reply:
> I can check the first element result[source].?[0] === ''?

Or length and first element?

Reply:
`result[source].?[0].length === 0`?

Reply:
I originally meant something like `result[source]?.length === 1 && result[source][0] === ''`, but whatever works

Reply:
most of them look like they have result[source]?.length === 1, updated the code